src/components/Register/NickName.tsx

```diff
@@ -75,15 +82,17 @@ const InputWrapper = styled.div`
   display: flex;
 `;
 
-const Button = styled.button`
+const Button = styled.button<{ $isEnabled: boolean }>`
   width: 6.4rem;
   height: 4.8rem;
   margin-left: 1rem;
 
   border: none;
   border-radius: 0.6rem;
-  background-color: ${({ theme }) => theme.colors.codrive_green};
-  color: ${({ theme }) => theme.colors.gray900};
+  background-color: ${({ theme, $isEnabled }) =>
+    $isEnabled ? theme.colors.codrive_green : theme.colors.gray500};
+  color: ${({ theme, $isEnabled }) =>
+    $isEnabled ? theme.colors.gray900 : theme.colors.white};
```

Comment:
P5: 아래처럼 한 번에 표현할 수도 있어요!

```suggestion
${({ theme, $isEnabled }) => 
   $isEnabled 
   ? css` 
      background-color: ${theme.colors.codrive_green};
      color: ${theme.colors.gray900};
      `
   : css` 
      background-color: ${theme.colors.gray500};
      color: ${theme.colors.white};
      `};
```